### PR TITLE
[Project 5] James Thomas Vertex Shaders

### PR DESCRIPTION
**Pull request by tho14063**
Finished the transformations.
lab05pic

### CODE REVIEW

File: definitions.h [hunk 1]
```diff
@@ -45,6 +47,261 @@ struct Vertex
...
 };
 
+
+/****************************************************
+ * Matrix for transforming vertices. 
+ ****************************************************/
+class Matrix
+{
+  private:
+    double * matrixPtr;
```
Comment (Napler) [inline definitions.h]: Very good way to make the matrix any size

File: definitions.h [hunk 2]
```diff
@@ -45,6 +47,261 @@ struct Vertex
...
+            delete [] matrixPtr;
+            matrixPtr = NULL;
+        }
+    }
+
+    /*************************************************************************
+     * Add Rotation
+     * Add rotation to the current matrix.
+     ************************************************************************/
+    void addRotation(double rot/*radians*/)
```
Comment (Napler) [inline definitions.h]: You'll eventually want to rotate on all axes.

File: definitions.h [hunk 3]
```diff
@@ -45,6 +47,261 @@ struct Vertex
     double w;
 };
 
+
```
Comment (Prestonh91) [inline definitions.h]: Great use of the Matrix class and 1D arrays

File: coursefunctions.h [hunk 4]
```diff
@@ -425,42 +425,64 @@ void TestVertexShader(Buffer2D<PIXEL> & target)
...
+	// DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, NULL);
 
         /******************************************************************
 		 * TRANSLATE (move +100 in the X direction, +50 in the Y direction)
          *****************************************************************/
         // Your translating code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
-
-		DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
+        Matrix transMatrix(4, 4);
+        transMatrix.addTranslation(100, 50, 0);
```
Comment (Prestonh91) [inline coursefunctions.h]: Smart use of the class and functions.

**Pull request merged**